### ISSUE TITLE
How to run `crate_universe` on the current crate.

### ISSUE THREAD

**Issue by sayrer** [bazelbuild/rules_rust#658]
One thing that similar tools do (e.g. Gazelle, https://github.com/bazelbuild/bazel-gazelle) is let you run the Bazel conversion on the package you're working in. I haven't seen an example of that for `crate_universe` or `cargo-raze`.

**Comment by UebelAndre** [comment]
FWIW: `cargo-raze` has [Using cargo-raze Through Bazel](https://github.com/google/cargo-raze/tree/c748b0c4ed7ad6b516f7838a27fee27a9d9c8664#using-cargo-raze-through-bazel)

**Comment by sayrer** [comment]
Right, I use that already, but it doesn't create a `BUILD.bazel` file next to `src/lib.rs`, for example.

**Comment by UebelAndre** [comment]
> Right, I use that already, but it doesn't create a `BUILD.bazel` file next to the local crate's `src/lib.rs`, for example.

Ah, I see now. https://github.com/google/cargo-raze/issues/339 might be the thing you're asking for? But I do not believe either `cargo-raze` or `crate_universe` can be used to initialize a new workspace or "Bazelify" an existing Cargo target.

**Comment by sayrer** [comment]
I was looking at converting this project, which has ~33 subcrates: https://github.com/linkerd/linkerd2-proxy/tree/main/linkerd. I suppose I could make a separate crate that imports it, and then convert things, but that seems pretty indirect.

**Comment by sayrer** [comment]
This is fixed by the `cargo_toml_files` attribute.

**Comment by illicitonion** [comment]
This currently isn't supported by `crate_universe`, but it should be pretty easy to add as a run-target. For now, though, having a Cargo.toml which depends on it will generate you the BUILD files you can copy and paste, but I agree it's indirect and not ideal!
